services/worker/src/worker/job_runners/parquet_and_dataset_info.py

```diff
@@ -503,7 +503,7 @@ def _request_size(url: str, hf_token: Optional[str] = None) -> Optional[int]:
     return int(size) if size is not None else size
 
 
-class _MockStreamingDownloadManager(StreamingDownloadManager):
+class _MockStreamingDownloadManager(StreamingDownloadManager):  # type: ignore
```

Comment:
not sure, but does it mean that mypy will totally ignore that class? If so, I would prefer to keep the previous way, with exceptions only for three lines, instead of a global exception for the class

Reply:
@severo please note this is needed because of `datasets` has become untyped.

You can check it:
- Just before I fix any mypy-related issue, in my second commit named "[Update poetry lock file](https://github.com/huggingface/datasets-server/pull/829/commits/993e6c7ce0012096c9c1054e138e53edf3ff2a04)", the CI already failed for this code line
  - CI: https://github.com/huggingface/datasets-server/actions/runs/4243203372/jobs/7375645159
  - Error:
    > src/worker/job_runners/parquet_and_dataset_info.py:506: error: Class cannot subclass "StreamingDownloadManager" (has type "Any")  [misc]
    As `datasets` is untyped, the `StreamingDownloadManager` has type `Any` and mypy does not allow to subclass from type `Any`

Reply:
> @severo please note this is needed because datasets has become untyped.

yes, I hadn't understood that. Therefore, we might have more things to change in the code? I'm not sure how mypy manages untyped, or partly typed, dependencies

Reply:
Also note that the "previous way" is not just ignoring the three lines above. If you look at the CI link above, you see that once `datasets` is untyped, we need:
- either adding the ignore comment to ALL datasets imports
- or instead (and equivalently), setting to ignore all `datasets` in the config file

Reply:
Yes, that's what I fear. But why is the CI working, if we are importing `datasets` everywhere? Is it because some of the methods are typed, but not others like `StreamingDownloadManager`?

Reply:
This specific mypy error is because we have `disallow_subclassing_any` enabled (by `strict = True` in the config), that is we tell mypy to raise an error if we subclass from a type `Any`.

Reply:
To avoid this general setting to be applied to this specific case, we need to put the `# type: ignore` comment here.

Reply:
OK!